lib/docker/image.rb

```diff
@@ -133,27 +138,36 @@ def get(id, opts = {}, conn = Docker.connection)
     # representation of the binary data. If the filename is not nil, then
     # return nil.
     def save(names, filename = nil, conn = Docker.connection)
-      # By using compare_by_identity we can create a Hash that has
-      # the same key multiple times.
-      query = {}
-      query.compare_by_identity
-      Array(names).each do |name|
-        query['names'.dup] = URI.encode(name)
-      end
-
       if filename
-        file = File.open(filename, 'wb')
-        conn.get(
-          '/images/get', query,
-          :response_block => response_block_for_save(file)
-        )
-        file.close
+        File.open(filename, 'wb') do |file|
+          save_stream(names, {}, conn, &response_block_for_save(file))
+        end
         nil
       else
-        conn.get('/images/get', query)
+        string = ''
```

Comment:
I would have used a `StringIO` here, but I had a hard time getting it to play nice with binary data.